### PR TITLE
ENH: Improve tanh for architectures without efficient gather/scatter …

### PR DESCRIPTION
**Pull request by Developer-Ecosystem-Engineering**
NumPy implements _tanh_ with a lookup table that isn't setup well for Apple silicon.  Transposing the lookup table makes it more efficient to load all coefficients of the polynomial.

- float32: 1.8x faster
- float64: 1.3x faster

**Apple M1 native (arm64):**
```
       before           after         ratio
     [7c143834]       [c3762e7a]
     <main>           <tanh/upstream-pr>
-         564±2μs          491±2μs     0.87  bench_ufunc.UFunc.time_ufunc_types('tanh')
-      410±0.04μs       324±0.03μs     0.79  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 1, 1, 'd')
-       429±0.2μs        336±0.3μs     0.78  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 1, 2, 'd')
-      450±0.05μs        350±0.1μs     0.78  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 2, 2, 'd')
-       452±0.3μs        352±0.8μs     0.78  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 4, 2, 'd')
-       432±0.5μs        335±0.5μs     0.78  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 2, 1, 'd')
-       435±0.8μs          337±1μs     0.77  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 4, 1, 'd')
-       466±0.2μs        360±0.5μs     0.77  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 2, 4, 'd')
-       444±0.3μs        343±0.3μs     0.77  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 1, 4, 'd')
-       467±0.3μs        359±0.3μs     0.77  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 4, 4, 'd')
-       237±0.1μs       147±0.03μs     0.62  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 4, 4, 'f')
-       230±0.4μs        143±0.2μs     0.62  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 4, 2, 'f')
-       224±0.1μs        138±0.8μs     0.61  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 2, 4, 'f')
-       199±0.2μs          122±1μs     0.61  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 1, 2, 'f')
-       216±0.6μs        131±0.2μs     0.61  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 2, 2, 'f')
-       208±0.4μs        125±0.3μs     0.60  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 1, 4, 'f')
-       203±0.3μs        120±0.9μs     0.59  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 2, 1, 'f')
-       216±0.7μs       125±0.03μs     0.58  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 4, 1, 'f')
-      190±0.04μs       110±0.06μs     0.58  bench_ufunc_strides.Unary.time_ufunc(<ufunc 'tanh'>, 1, 1, 'f')
```

### CODE REVIEW

File: numpy/core/src/umath/loops_hyperbolic.dispatch.c.src
Comment (mattip) [inline numpy/core/src/umath/loops_hyperbolic.dispatch.c.src:92]: I don't see any actual use of `npyv_get_lane_u64` in the code.

Comment (Developer-Ecosystem-Engineering) [reply numpy/core/src/umath/loops_hyperbolic.dispatch.c.src]: Removed thanks!

**Comment by rgommers** [comment]
The one CI failure is clearly unrelated:
```
/home/circleci/repo/doc/source/release/{version}-notes.rst: WARNING: document isn't included in any toctree
```

**Review by seiko2plus — APPROVED** [review]
LGTM, Thank you! After merging, I will cover vector lane extractions for the other SIMD extensions.

**Comment by seberg** [comment]
Looks fine to me as well, and if Sayed is happy this is good to go in.  Thanks @Developer-Ecosystem-Engineering.

**Pull request merged**
